### PR TITLE
Establish recursive upload of directories

### PR DESCRIPTION
**Pull request by PhilippRo**
Now the command "Upload Scripts From Folder" of the context menu is also able to upload scripts in subdirectories.

At the moment it is not able to put scripts from subdirectories into subdirectories.

Solves issue #72

### CODE REVIEW

**Comment by bkircher** [comment]
@PhilippRo Can you add an entry to the Changelog and can you add the commit to this PR?

**Comment by PhilippRo** [comment]
This will be done after m-Schlitzer updated the Changelog in master to avoid merge conflicts

**Comment by bkircher** [comment]
:+1:

File: src/commands.ts [hunk 1]
```diff
@@ -186,6 +185,7 @@ export async function uploadAll(loginData: nodeDoc.LoginData, paramFolderName: s
     } catch (reason) {
         vscode.window.showErrorMessage('upload all failed: ' + reason);
     }
+    vscode.window.setStatusBarMessage('uploaded all scripts successfully');
```
Comment (tineheller) [inline src/commands.ts]: I think this line should be inserted after the `await promise` command.

**Comment by FabianTe** [comment]
Is this tested already? Then a merge would be awesome, I'd love to have this feature ;)

**Comment by PhilippRo** [comment]
Uploading works, but the whole feature is currently not finished, because it ignores the conflict-mode. The conflict-mode is supposed to prohibit overwriting files that had been changed on the server manually. Might be done in a couple of weeks.

**Comment by tineheller** [comment]
Actually the recursive upload should be implemented in [node-documents-scripting](https://github.com/otris/node-documents-scripting). @wehrstedt added the function `readDirSync` to that module. `readDirSync` returns a list of all scripts from a folder and subfolders. So the `uploadAll` function simply has to be called with list returned by `readDirSync`.

**Comment by tineheller** [comment]
The recursive upload is resolved with 3b9ae92aff44f2156943dd8e65287a792959f94f by simply calling the new function `getScriptsFromFolderSync()` instead of `getScriptsFromFolder()` in `node-documents-scripting`.

**Pull request closed without merging**